### DISCUSSION TITLE
Why importing from the same file?

### DISCUSSION THREAD

**Discussion by zeritonius** [vuejs/petite-vue#157] [General]
What are the benefits of importing from the same file, please? I'm not familiar with this approach.

Example:
[https://github.com/vuejs/petite-vue/blob/main/src/directives/effect.ts](https://github.com/vuejs/petite-vue/blob/main/src/directives/effect.ts)

```
import { Directive } from '.'
import { execute } from '../eval'
import { nextTick } from '../scheduler'

export const effect: Directive = ({ el, ctx, exp, effect }) => {
  nextTick(() => effect(() => execute(ctx.scope, exp, el)))
}
```

Thank you.

**Comment by halivert** [comment]
It's not importing from the same file, but from the same directory, so that import could also look like:
```ts
import { Directive } from './index'
```
but shorter

**Reply by zeritonius** [reply]
Thank you so much for clarifying this.
